src/main/java/com/fasterxml/jackson/core/util/BufferRecyclerProvider.java

```diff
@@ -0,0 +1,19 @@
+package com.fasterxml.jackson.core.util;
+
+import com.fasterxml.jackson.core.TokenStreamFactory;
+
+/**
+ * Provider for {@link BufferRecycler}s to allow pluggable and optional
+ * recycling of underlying input/output buffers.
+ *
+ * @since 2.16
+ */
+public abstract class BufferRecyclerProvider
+    implements java.io.Serializable
+{
+    private static final long serialVersionUID = 1L;
+
+    public abstract BufferRecycler acquireBufferRecycler(TokenStreamFactory forFactory);
```

Comment:
Wasn't sure of naming for get/return etc, open to different names than acquire/release.

Reply:
Naming wise I believe that what's wrong here is the name of the class. The fact that is a pool should be reflected in its name, so we should call it `BufferRecyclerPool`. Once you do this `acquire` and `release` will be the most natural choices, but of course I'm open to other suggestions.

Reply:
Hmmmh. Ok. Not sure how I managed to miss this comment before merging.

I agree that if there's life-cycle, "provider" is not quite the right choice.
Not sure I like "pool" but maybe that'd be better...
Let me first merge things as they are and the consider renaming.